pkg/chunk/local/fs_object_client.go

```diff
@@ -81,6 +96,58 @@ func (f *FSObjectClient) getChunk(_ context.Context, decodeContext *chunk.Decode
 	return c, nil
 }
 
+// Get object from the store
+func (f *FSObjectClient) GetObject(ctx context.Context, objectKey string) (io.ReadCloser, error) {
+	return os.Open(path.Join(f.cfg.Directory, objectKey))
+}
+
+// Put object into the store
+func (f *FSObjectClient) PutObject(ctx context.Context, objectKey string, object io.ReadSeeker) error {
+	fullPath := path.Join(f.cfg.Directory, objectKey)
+	err := util.EnsureDirectory(path.Dir(fullPath))
+	if err != nil {
+		return err
+	}
+
+	fl, err := os.OpenFile(fullPath, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0644)
+	if err != nil {
+		return err
+	}
+
+	defer fl.Close()
+
+	_, err = io.Copy(fl, object)
+	return err
+}
+
+// List objects from the store
+func (f *FSObjectClient) List(ctx context.Context, prefix string) (map[string]time.Time, error) {
+	folderPath := filepath.Join(f.cfg.Directory, prefix)
+
+	_, err := os.Stat(folderPath)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return nil, nil
+		}
+		return nil, err
+	}
+
+	filesInfo, err := ioutil.ReadDir(folderPath)
```

Comment:
This is not recursive. Just to confirm: is it what we want? I would document it in the function comment.